appengine/pubsub/app.js

```diff
@@ -70,9 +74,37 @@ app.post('/', formBodyParser, async (req, res, next) => {
 // [END gae_flex_pubsub_index]
 
 // [START gae_flex_pubsub_push]
-app.post('/pubsub/push', jsonBodyParser, (req, res) => {
+app.post('/pubsub/push', jsonBodyParser, async (req, res) => {
+  // Verify that the request originates from the application.
   if (req.query.token !== PUBSUB_VERIFICATION_TOKEN) {
-    res.status(400).send();
+    res.status(400).send('Invalid request');
+    return;
+  }
+
+  // Verify that the push request originates from Cloud Pub/Sub.
+  try {
+    // Get the Cloud Pub/Sub-generated JWT in the "Authorization" header.
+    const bearer = req.header('Authorization');
+    const token = bearer.split(' ').pop();
+    tokens.push(token);
+
+    // Verify and decode the JWT.
+    const ticket = await authClient.verifyIdToken({
+      idToken: token,
+      audience: 'example.com',
+    });
+    const claim = ticket.getPayload();
+
+    // Must also verify the `iss` claim
```

Comment:
Could you confirm that the Node `verifyIdToken` function verifies the JWT signature, the `aud` claim, the `exp` claim, and the `iss` claim? (The Python equivalent doesn't verify `iss` thus this block of code in the Python sample).

Reply:
Oh! Good call, looks like the client verifies all of those fields (via [verifySignedJwtWithCertsAsync](https://github.com/googleapis/google-auth-library-nodejs/blob/master/src/auth/oauth2client.ts#L1019)), I'll remove this part!

Reply:
Done!